src/main/java/com/faforever/client/vault/VaultFileSystemLocationChecker.java

```diff
@@ -0,0 +1,62 @@
+package com.faforever.client.vault;
+
+import com.faforever.client.i18n.I18n;
+import com.faforever.client.notification.Action;
+import com.faforever.client.notification.NotificationService;
+import com.faforever.client.notification.PersistentNotification;
+import com.faforever.client.notification.Severity;
+import com.faforever.client.preferences.ForgedAlliancePrefs;
+import com.faforever.client.preferences.PreferencesService;
+import com.google.common.base.CharMatcher;
+import com.sun.security.auth.module.NTSystem;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.regex.Pattern;
+
+import static org.bridj.Platform.isUnix;
+import static org.bridj.Platform.isWindows;
+
+/**
+ * Checks if the user is on linux, has a none ascii username or has One Drive installed. In this cases we need to use a fallback location for the
+ * vault.
+ */
+@Component
+public class VaultFileSystemLocationChecker {
+  public static final Pattern ONE_DRIVE_FOLDER_NAME_IDENTIFIER = Pattern.compile("[Oo]ne[ ]*[Dd]rive");
```

Comment:
Don't we wanna fallback on non ascii name chars aswell?

Reply:
[ ]* can be replace by *

Reply:
Whatever @Brutus5000 did that and I think it is better to read like this